BuildTasks/Common/Common.ts

```diff
@@ -150,6 +150,26 @@ export function runTfx(cmd: (tfx: ToolRunner) => void) {
 }
 
 /**
+ * Reads the extension version from the supplied variable
+ *
+ * @param  {string="connectedServiceName"} inputFieldName
+ * @returns string
+ */
+export function getExtensionVersion(): string {
+    const extensionVersion = tl.getInput("extensionVersion", false);
+    if (extensionVersion) {
+        const extractedVersions = extensionVersion.match(/[0-9]+\.[0-9]+\.[0-9]+(\.[0-9]+)?/);
+        if (extractedVersions && extractedVersions.length === 1) {
+            return extractedVersions[0];
+        }
+        else {
+            tl.error(`Supplied ExtensionVersion must contain a string matching '##.##.##'.`);
```

Comment:
Shouldn't we throw? What  happens if we can't extracted the version? For example, does the Package/Publish extension succeed using the version declared in the manifest, even if we are emitting a `tl.error`?